internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -336,6 +337,31 @@ func (r *Reconciler) machineDeploymentToCluster(o client.Object) []ctrl.Request
 	}}
 }
 
+func (r *Reconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Cluster) (ctrl.Result, error) {
+	// Add the `ok-to-delete` annotation if the BeforeClusterDelete hook returns a successful non-blocking response.
+	// Adding this annotation will trigger the cluster reconciler to proceed with deleting the cluster.
```

Comment:
Nit: this feels more like a comment for the part were we set the annotation. Can we reframe a bit to start with That we are calling the hook and if successful add the ...